tools/stats/upload_external_contrib_stats.py

```diff
@@ -126,17 +127,35 @@ def get_external_pr_data(
     )
     args = parser.parse_args()
     for i in range(args.length):
+        tries = 0
         startdate = args.startDate + datetime.timedelta(days=i)
         data = get_external_pr_data(
             startdate,
             startdate + datetime.timedelta(days=args.period_length),
             period_length=args.period_length,
         )
-        upload_to_s3(
-            bucket_name="torchci-contribution-data",
-            key=f"external_contribution_counts/{str(startdate)}",
-            docs=data,
-        )
+        while tries < MAXIMUM_RETRIES:
```

Comment:
Do you have any examples at hand where this is flaky?  It comes as a surprise to me as the connection from the runner (ec2) to S3 is local and should be very stable